### PR TITLE
Show correct icons in remote object inspector (EditorDebuggerRemoteObject)

### LINKED ISSUE #95209: Remote Scene doesn't use C# IconAttribute for node instanciated, but Inspector is able to use it?
### Tested versions

Probably reproductible on all version.

### System information

Window 10 - 4.3-RC2

### Issue description

I'm working with Godot C# and I use IconAttribute to have custom icon in the editor. `[GlobalClass, Icon("res://assets/textures/icons/tbd-icon.svg")]` Mostly everywhere you will be able to see the icon, but in the "Remote Scene" the tree view won't show any custom icon for instanciated node. You click on the node which update the Inspector view and you will see the EditorDebuggerRemoteObject do include the CustomIcon but not the Remote Scene 

image

### Steps to reproduce

1. This
```csharp
using System;
using Godot;

[GlobalClass, Icon("res://path/to/icon/some-icon.svg")]
public partial class GlobalManager : Node3D, IGlobalManager {
	public static GlobalManager Instance { get; private set; }
}
```
2. Make it Autoload or just instanciate in a working scene. 
3. Run the scene in the debugger and pause (or not!) and look at the Remote Scene tree where custom icon won't be use. 
4. For reference, I use this site (https://uxwing.com/) for icons, and edit the SVG for better representation of the icon. 

### Minimal reproduction project (MRP)

[mrp-custom-icon.zip](https://github.com/user-attachments/files/16514272/mrp-custom-icon.zip)

### PR DESCRIPTION
**Pull request by jaydensipe**
~~Used to implement https://github.com/godotengine/godot-proposals/issues/2084, used to fix #95209.~~ <- Better implemented in https://github.com/godotengine/godot/pull/88427.

~~This commit allows the remote scene view to display custom icons from classes (annotated with `@icon`) and allows the remote object inspector to display the correct class icon for an Object. These help a lot with readability.~~

This commit allows the remote object inspector (EditorDebuggerRemoteObject) to display the correct class icon for an Object.

---
| Before | After |
| ------------- | ------------- |
| <video src=https://github.com/user-attachments/assets/62f33b1f-942b-4e6c-b66f-7ccfce4cdb94></video> | <video src=https://github.com/user-attachments/assets/ba8d50b9-6613-4d75-833c-7deb3ae912b9></video>

### CODE REVIEW

File: scene/debugger/scene_debugger.cpp [hunk 1]
```diff
@@ -535,7 +535,16 @@ SceneDebuggerTree::SceneDebuggerTree(Node *p_root) {
 				}
 			}
 		}
-		nodes.push_back(RemoteNode(count, n->get_name(), n->get_class(), n->get_instance_id(), n->get_scene_file_path(), view_flags));
+
+		String class_name;
+		if (Ref<Script> script_base = n->get_script(); script_base.is_valid()) {
```
Comment (timothyqiu) [inline scene/debugger/scene_debugger.cpp:540]: We don't use init statement syntax in the codebase.

**Review by timothyqiu — COMMENTED** [review]
I think you can merge #98157 into this one. It is essentially the same as this one.

File: scene/debugger/scene_debugger.cpp (hunk 1 above)
Comment (jaydensipe) [reply scene/debugger/scene_debugger.cpp]: Updated.

**Comment by timothyqiu** [comment]
I set the node's icon via `@icon`:

```gdscript
@icon("res://icon.svg")
extends Node2D
```

| | Before | After | Expected |
|---|---|---|---|
| Remote Scene Tree | No Icon | No Icon | Icon |
| Remote Inspector   | Icon        | No Icon | Icon |

Project: [test.zip](https://github.com/user-attachments/files/17357809/test.zip)

**Comment by jaydensipe** [comment]
> I set the node's icon via `@icon`:
> 
> ```gdscript
> @icon("res://icon.svg")
> extends Node2D
> ```
> 
> Before	After	Expected
> Remote Scene Tree	No Icon	No Icon	Icon
> Remote Inspector	Icon	No Icon	Icon
> Project: [test.zip](https://github.com/user-attachments/files/17357809/test.zip)

So, I did a lot of looking into this; to fix your test you can declare a class_name like so, and it will work: (BUT...)
> ```gdscript
> @icon("res://icon.svg")
> extends Node2D
> class_name Test2D
> ```

I just found out there was another PR (https://github.com/godotengine/godot/pull/88427) that implemented this... oops :P. Theirs works without declaring `class_name` so depending on if we want to merge that one or I can work on this one to implement that. Theirs also doesn't implement the EditorDebuggerRemoteObject fixes, so I can wait until theirs gets merged to use some of their code to implement that better.

**Comment by CrayolaEater** [comment]
If implementation wise your fix is similar to mine than I don't think having 2 open PRs that solve the same problem is a good idea. Maybe create a PR only for the EditorDebuggerRemoteObject fix?

**Comment by jaydensipe** [comment]
> If implementation wise your fix is similar to mine then I don't think having 2 open PRs that solve the same problem is a good idea. Maybe create a PR only for the EditorDebuggerRemoteObject fix?

Agreed, changed this PR into the **EditorDebuggerRemoteObject fix**, but if we need a new PR for some reason I can do that. I tested my changes with yours, and they worked good. BTW, your fix works for Autoload singleton scripts which is awesome and will help a ton with my projects.

**Comment by timothyqiu** [comment]
Currently, the node's icon is completely missing in the remote object inspector.

Using the test project from https://github.com/godotengine/godot/pull/98156#issuecomment-2409499792

**Comment by jaydensipe** [comment]
> Currently, the node's icon is completely missing in the remote object inspector.
> 
> Using the test project from [#98156 (comment)](https://github.com/godotengine/godot/pull/98156#issuecomment-2409499792)

Hey, https://github.com/godotengine/godot/pull/88427 has code in it that would fix the icon missing when not defining a `class_name`, so would we wait until that is merged to merge this?

**Review by timothyqiu — APPROVED** [review]
Works as expected with #88427 which should be merged first.

File: editor/gui/editor_object_selector.cpp [hunk 2]
```diff
@@ -131,6 +132,19 @@ void EditorObjectSelector::update_path() {
 		Ref<Texture2D> obj_icon;
 		if (Object::cast_to<MultiNodeEdit>(obj)) {
 			obj_icon = EditorNode::get_singleton()->get_class_icon(Object::cast_to<MultiNodeEdit>(obj)->get_edited_class_name());
+		} else if (obj->is_class("EditorDebuggerRemoteObject")) {
```
Comment (KoBeWi) [inline editor/gui/editor_object_selector.cpp:135]:
```suggestion
		} else if (Object::cast_to<EditorDebuggerRemoteObject>(obj)) {
```

**Review by KoBeWi — APPROVED** [review]

**Pull request merged**